sdks/python/apache_beam/runners/portability/abstract_job_service.py

```diff
@@ -16,16 +16,36 @@
 #
 from __future__ import absolute_import
 
+import itertools
 import logging
 import uuid
 from builtins import object
 
+from google.protobuf import timestamp_pb2
+
 from apache_beam.portability.api import beam_job_api_pb2
 from apache_beam.portability.api import beam_job_api_pb2_grpc
+from apache_beam.utils import proto_utils
+from apache_beam.utils.timestamp import Timestamp
 
 _LOGGER = logging.getLogger(__name__)
 
 
+def make_state_event(state, timestamp):
+  if isinstance(timestamp, Timestamp):
+    proto_timestamp = timestamp.to_proto()
+  elif isinstance(timestamp, timestamp_pb2.Timestamp):
+    proto_timestamp = timestamp
+  elif isinstance(timestamp, float):
```

Comment:
nit: I thought you would have removed this float -> proto conversion and users would always use Beam's Timestamp.

Reply:
Fixed.   At the time I thought it could be useful for an end-user implementing their own job service, but it's not too much to ask call `Timestamp.now()`.